src/main/java/org/lightcouch/CouchDbProperties.java

```diff
@@ -192,4 +195,27 @@ public boolean isSSLAuthenticationDisabled() {
 		return disableSSLAuthentication;
 	}
 
+	/**
+	 * Returns the SSLSocketFactory that gets used when connecting to
+	 * CouchDB over a <code>https</code> URL.
+	 * @return An SSLSocketFactory, or <code>null</code>, which stands for
+	 *         the default SSLSocketFactory of the JRE.
+	 * @see #setSecureSSLSocketFactory(javax.net.ssl.SSLSocketFactory)
+	 */
+	public SSLSocketFactory getSecureSSLSocketFactory() {
+		return secureSSLSocketFactory;
+	}
+
+	/**
+	 * Specifies the SSLSocketFactory to use when connecting to CouchDB
+	 * over a <code>https</code> URL.
+	 * @param factory An SSLSocketFactory, or <code>null</code> for the
+	 *                default SSLSocketFactory of the JRE.
+	 * @see #getSecureSSLSocketFactory()
+	 */
+	public CouchDbProperties setSecureSSLSocketFactory(SSLSocketFactory factory) {
+		this.secureSSLSocketFactory = factory;
+		return this;
+	}
```

Comment:
> Why are these called "setSecureSSLSocketFactory" and not just setSSLSocketFactory?
The name 'setSSLSocketFactory' would imply that this SSLSocketFactory is used always. But we don't want it to be used in the "insecure" mode - because the use-case of the custom SSLSocketFactory is to increase security (e.g. by using only the TLSv1.2 protocol, or by using a truststore that contains very few certificate authorities), and the point of the insecure mode is to avoid all kinds of SSL configuration hassles in situations where security does not matter.

> I'm not sure the socket factory that is set need be more secure...
It does not need to be. The method name should not imply "set a more secure socket factory" but is meant to imply "set a socket factory for the secure mode".